logicmonitor/resource_collector_group.go

```diff
@@ -100,24 +123,86 @@ func updateCollectorGroup(d *schema.ResourceData, m interface{}) error {
 		d.SetPartial(v)
 	}
 
+	log.Printf("payload response %v", restCollectorGroupResponse.Payload)
 	d.Partial(false)
 	return nil
 }
 
 // function to delete collector groups
 func deleteCollectorGroup(d *schema.ResourceData, m interface{}) error {
-	client := m.(*lmv1.DefaultApi)
-	groupID, err := strconv.Atoi(d.Id())
+	client := m.(*lmclient.LMSdkGo)
+	id, err := strconv.Atoi(d.Id())
 	if err != nil {
 		return err
 	}
-	restCollectorGroupResponse, apiResponse, e := client.DeleteCollectorGroupById(int32(groupID))
-	err = checkStatus(restCollectorGroupResponse.Status, restCollectorGroupResponse.Errmsg, apiResponse.StatusCode, apiResponse.Status, e)
+
+	params := lm.NewDeleteCollectorGroupByIDParams()
+	params.ID = int32(id)
+	restCollectorGroupResponse, err := client.LM.DeleteCollectorGroupByID(params)
+
 	if err != nil {
 		log.Printf("Error deleting collector group %s", err)
 		d.SetId("")
 	}
 
+	log.Printf("payload response %v", restCollectorGroupResponse.Payload)
 	d.SetId("")
 	return nil
 }
+
+func resourceCollectorGroupStateImporter(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	client := m.(*lmclient.LMSdkGo)
+
+	// if it is a groupId, we will add the group directly
+	if checkID(d.Id()) {
+		id, err := strconv.Atoi(d.Id())
+		if err != nil {
+			return nil, err
+		}
+		params := lm.NewGetCollectorGroupByIDParams()
+		params.SetID(int32(id))
+		restCollectorGroupResponse, err := client.LM.GetCollectorGroupByID(params)
+		if err != nil {
+			log.Printf("Failed to find collector group %q", err)
+			return nil, err
+		}
+		d.Set("name", restCollectorGroupResponse.Payload.Name)
+		d.Set("description", restCollectorGroupResponse.Payload.Description)
+
+		properties := make(map[*string]*string)
+		props := restCollectorGroupResponse.Payload.CustomProperties
+		for _, v := range props {
+			properties[v.Name] = v.Value
+		}
+		d.Set("properties", properties)
+		d.SetId(d.Id())
+	} else {
+
+		// finding collector group by name
+		filter := fmt.Sprintf("name:\"%s\"", d.Id())
+		params := lm.NewGetCollectorGroupListParams()
+		params.SetFilter(&filter)
+
+		//looks for collector group
+		restCollectorGroupPaginationResponse, err := client.LM.GetCollectorGroupList(params)
+		if err != nil {
+			return nil, err
+		}
+
+		if restCollectorGroupPaginationResponse.Payload.Total > 0 {
```

Comment:
there's no guarantee that filter will return one and only one collector group. you should also error if Total > 1 with an appropriate message. Otherwise, you risk importing the wrong group due to an ambiguous user input.

Reply:
sounds good, i can do an if greater than 1 error out, else if 0, set the ids, else error out